pkg/proc/native/proc.go

```diff
@@ -52,12 +54,30 @@ func newProcess(pid int) *nativeProcess {
 		os:             new(osProcessDetails),
 		ptraceChan:     make(chan func()),
 		ptraceDoneChan: make(chan interface{}),
+		ptraceRefCnt:   new(int),
 		bi:             proc.NewBinaryInfo(runtime.GOOS, runtime.GOARCH),
 	}
+	(*dbp.ptraceRefCnt)++
 	go dbp.handlePtraceFuncs()
 	return dbp
 }
 
+// newChildProcess is like newProcess but uses the same ptrace thread as dbp.
+func newChildProcess(dbp *nativeProcess, pid int) *nativeProcess {
+	(*dbp.ptraceRefCnt)++
+	return &nativeProcess{
+		pid:            pid,
+		threads:        make(map[int]*nativeThread),
+		breakpoints:    proc.NewBreakpointMap(),
+		firstStart:     true,
+		os:             new(osProcessDetails),
+		ptraceChan:     dbp.ptraceChan,
```

Comment:
Perhaps we should just store a reference to the root process instead of copying over individual fields? We could hide the reference behind an interface to limit what can be accessed.

Reply:
If we add a `parent *nativeProcess` field then every time we need to reference these three fields we have to do `if dbp.parent != nil { dbp.parent.ptraceChan... } else { dbp.ptraceChan... }` which is weird.

We could add a type like this:

```
type ptraceThread struct {
   ptraceChan chan func()
   ptraceDoneChan chan interface{}
   ptraceRefCnt int
}
```

and then keep a pointer to it in nativeProcess, which parents and children would share.

Or were you thinking something else?

Reply:
Yes, that solution is along the lines of what I had in mind.

Reply:
Done.